app/models/discovery_rule.rb

```diff
@@ -36,13 +36,31 @@ class DiscoveryRule < ApplicationRecord
     end
   }
 
+  def unique_priority
```

Comment:
Does this validation pass for a discovery_rule in case of update?

With the current code, if you try to update a discovery_rule record without change in priority column, it will definitely fail every time.
This is one of reason why you are getting errors for few tests.

Reply:
Good catch @kgaikwad :) thanks. Will make the corrections.